### PR TITLE
send product_id attribute with key transaction events

### PR DESCRIPTION
**Pull request by cavvia**
Our third party marketing channels need to track conversion events relating to transactions. For this purpose, we need to attach a `product_id` attribute to these events. This should be the mongo ID of the Artwork in question.

See also artsy/force#3870

### CODE REVIEW

File: src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx [hunk 1]
```diff
@@ -47,6 +47,7 @@ export class ArtworkSidebarBidAction extends React.Component<
 
   @track((props: ArtworkSidebarBidActionProps) => ({
     artwork_slug: props.artwork.id,
+    product_id: props.artwork.id,
```
Comment (cavvia) [inline src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarBidAction.tsx]: I have an issue here in that the `ArtworkSidebarBidAction_artwork` representation only has a slug but no Mongo ID declared on it. How do I regenerate the graphql definition for the Artwork here? I need to pull in `_id` on the artwork.

File: src/Apps/Order/Routes/Review/index.tsx [hunk 2]
```diff
@@ -51,14 +51,15 @@ export class ReviewRoute extends Component<ReviewProps> {
         ? Schema.ActionType.SubmittedOrder
         : Schema.ActionType.SubmittedOffer,
     order_id: props.order.id,
+    product_id: props.order.lineItems.edges[0].node.artwork.id
```
Comment (cavvia) [inline src/Apps/Order/Routes/Review/index.tsx]: I have the same issue here... the Mongo ID doesn't appear to be available in this artwork representation either.

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit fb9f60efc73ea840ca9ca215d3aed2e4b2ff6795

https://deploy-preview-2274--artsy-reaction.netlify.com

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 950f59e45a043ba8d16ebd1b713144a8135a93e8

https://deploy-preview-2274--artsy-reaction.netlify.com

File: src/__generated__/ArtworkSidebarBidAction_artwork.graphql.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export type ArtworkSidebarBidAction_artwork = {
         }) | null;
     }> | null;
     readonly id: string;
+    readonly _id: string;
```
Comment (alloy) [inline src/__generated__/ArtworkSidebarBidAction_artwork.graphql.ts]: Did you manually edit this?

File: src/__generated__/Review_order.graphql.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@ export type Review_order = {
             readonly node: ({
                 readonly artwork: ({
                     readonly id: string;
+                    readonly _id: string;
```
Comment (alloy) [inline src/__generated__/Review_order.graphql.ts]: Ditto

File: src/__generated__/ArtworkSidebarBidAction_artwork.graphql.ts (hunk 3 above)
Comment (cavvia) [reply src/__generated__/ArtworkSidebarBidAction_artwork.graphql.ts]: yes i manually edited this to see if i could get feedback on the correct process... will run the compiler and fix this up, thanks for the tip!

**Comment by cavvia** [comment]
@alloy Thanks! I've updated the relevant fragments (the `ArtworkSidebarBidAction_artwork` fragment is actually not the one you linked to but I looked it up) and ran `yarn relay`.

**Comment by alloy** [comment]
My bad, glad you got it figured out, though!

Looks like some test fixtures need to be updated after this change, I can take a stab at that if you’d like?

**Comment by cavvia** [comment]
I fixed up the fixture... all 🍏 now!

**Comment by alloy** [comment]
Also needs a rebase btw 🙈 

Merge on green.

**Comment by artsyit** [comment]
:rocket: PR was released in v15.14.2 :rocket:

**Pull request merged**